protocols/protocol.py

```diff
@@ -274,6 +289,12 @@ def validate_debug(cls, jsonDict, validation_result, expected_schema=None):
                             schema_type=f.type,
                             value=datum.get(f.name)
                         )
+            else:
```

Comment:
@priesgo  This will never be executed. Previously there is `if` `elif` statements with the same check.

Reply:
wait, it's the same code as line 225

Reply:
I have to disagree @gerlv, beware that that piece of new code is inside `schema_type in ['record', 'error', 'request']`
Furthermore, I added some tests to check this, you can remove that change and see the tests failing.